torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -308,20 +336,48 @@ def realize_tensor(t):
 
         mat1 = realize_tensor(mat1)
         mat2 = realize_tensor(mat2)
-        if op is torch.ops.aten.bmm or op is torch.ops.aten.mm:
-            ori_time = do_bench(
-                lambda: op(mat1, mat2),
-            )
-        else:
-            if input is not None:
-                input = realize_tensor(input)
-            ori_time = do_bench(
-                lambda: op(input, mat1, mat2),
-            )
+
+        # since we key on whether or not the inputs can be memory planned, set cache for the
+        # original time which is unaffected by whether or not the input can be planned
+        ori_time_key = "ori_time " + should_pad_bench_key(
+            match, mat1, mat2, op, input, ignore_should_pad=True
+        )
+        ori_time = get_cached_should_pad(ori_time_key)
+        if ori_time is None:
+            if op is torch.ops.aten.bmm or op is torch.ops.aten.mm:
+                ori_time = do_bench(
+                    lambda: op(mat1, mat2),
+                )
+            else:
+                if input is not None:
+                    input = realize_tensor(input)
```

Comment:
Explain a bit why a realize is needed here?

Reply:
hmm, I mean, why we need realize? lol

Reply:
realize just means instantiate here. I imagine `input` should never not be None here. or we should dispatch to mm. but will leave for follow up.